src/gui/guiFormSpecMenu.cpp

```diff
@@ -3138,6 +3140,120 @@ void GUIFormSpecMenu::acceptInput(FormspecQuitMode quitmode=quit_mode_no)
 			fields["quit"] = "true";
 		}
 
+		#define BTOS(b) (b ? "true" : "false")
+
+		if (current_keys_pending.pending_mouse_event) {
+			SEvent::SMouseInput &event = current_keys_pending.mouse_event;
+
+			std::string field_type = "error";
+
+			#define SET_MOUSE_EVENT(e, t)	\
```

Comment:
`DETECT_MOUSE_EVENT` might be a better name?

Reply:
Probably.  I need to rebase, so I might as well do this anyway.